packages/nuxt/src/app/plugins/async-data-shallow.ts

```diff
@@ -0,0 +1,9 @@
+import { defineNuxtPlugin } from '#app/nuxt'
+
+export default defineNuxtPlugin({
+  name: 'nuxt:async-data-shallow',
+  enforce: 'pre',
+  setup (nuxtApp) {
+    nuxtApp._asyncDataShallow = true
```

Comment:
can we use `nuxt.config.mjs` instead?

https://github.com/nuxt/nuxt/blob/a2fb8234ac629c67776f38b691fab273838ef48e/packages/nuxt/src/core/templates.ts#L338-L353

Reply:
let's defer this to another PR